src/fortran/client.F90

```diff
@@ -1444,4 +1464,259 @@ function use_tensor_ensemble_prefix(self, use_prefix) result(code)
+  dest_name_length = len_trim(dest_name)
+
+  code = copy_list_c(self%client_ptr, src_name_c, src_name_length, dest_name_c, dest_name_length)
+end function copy_list
+
+!> Rename an aggregation list
+function rename_list(self, src_name, dest_name) result(code)
+  class(client_type),   intent(in) :: self     !< An initialized SmartRedis client
+  character(len=*),     intent(in) :: src_name !< Name of the dataset to rename
+  character(len=*),     intent(in) :: dest_name !< The new list name
+  integer(kind=enum_kind)          :: code
+  
+  integer(kind=c_size_t) :: src_name_length, dest_name_length
+  character(kind=c_char,len=len_trim(src_name)) :: src_name_c
+  character(kind=c_char,len=len_trim(dest_name)) :: dest_name_c
+
+  src_name_c = trim(src_name)
+  src_name_length = len_trim(src_name)
+  dest_name_c = trim(dest_name)
+  dest_name_length = len_trim(dest_name)
+
+  code = rename_list_c(self%client_ptr, src_name_c, src_name_length, dest_name_c, dest_name_length)
+end function rename_list
+
+!> Get the length of the aggregation list
+function get_list_length(self, list_name, result_length) result(code)
+  class(client_type),   intent(in) :: self       !< An initialized SmartRedis client
+  character(len=*),     intent(in) :: list_name    !< Name of the dataset to get
+  integer,              intent(out) :: result_length !< The length of the list
+  integer(kind=enum_kind)          :: code
+ 
+  integer(kind=c_size_t) :: list_name_length 
+  integer(kind=c_int) :: result_length_c
+  character(kind=c_char,len=len_trim(list_name)) :: list_name_c
+
+  list_name_c = trim(list_name)
+  list_name_length = len_trim(list_name)
+
+  code = get_list_length_c(self%client_ptr, list_name_c, list_name_length, result_length_c)
+  result_length = result_length_c
+
+end function get_list_length
+
+!> Get the length of the aggregation list
+function poll_list_length(self, list_name, list_length, poll_frequency_ms, num_tries, poll_result) result(code)
+  class(client_type),   intent(in) :: self       !< An initialized SmartRedis client
+  character(len=*),     intent(in) :: list_name    !< Name of the dataset to get
+  integer,              intent(in)  :: list_length !< The desired length of the list
+  integer,              intent(in)  :: poll_frequency_ms  !< Frequency at which to poll the database (ms)
+  integer,              intent(in)  :: num_tries          !< Number of times to poll the database before failing
+  logical(kind=c_bool), intent(out) :: poll_result !< True if the list is the requested length, False if not after num_tries.
+  integer(kind=enum_kind)          :: code
+  
+  ! Local variables
+  character(kind=c_char, len=len_trim(list_name)) :: list_name_c
+  integer(kind=c_size_t) :: list_name_length
+  integer(kind=c_int) :: c_poll_frequency, c_num_tries, c_list_length
+
+  list_name_c = trim(list_name)
+  list_name_length = len_trim(list_name)
+  c_num_tries = num_tries
+  c_poll_frequency = poll_frequency_ms
```

Comment:
is this translation necessary? I don't know if we've done it consistently in other places

Reply:
(from `int` to `c_int`, that is)

Reply:
We talked about this offline, but for posterity: for the sake of absolute best practices, we should be much better about explicitly converting from the default integer type that the compiler uses to a specific c-integer. There are a number of cases where the users might specify that all integers/reals/etc. should 64-bit numbers where we might run into trouble if we pass them straight to the c client.